### DISCUSSION TITLE
Quarto PDF book: "LaTeX Error: Environment abstract undefined"

### DISCUSSION THREAD

**Discussion by isaactpetersen** [quarto-dev/quarto-cli#11193] [Q&A] [answered]
### Description

I posted this question on Posit forums a week ago but haven't received any responses (https://forum.posit.co/t/quarto-pdf-book-latex-error-environment-abstract-undefined/194119).

I am writing a quarto book. I include `abstract:` under `book:` in the `_quarto.yml` file. My GitHub Actions knits the file to several output formats (HTML, EPUB, and PDF). Knitting to HTML and EPUB works fine. However, knitting to PDF errors out with the following error:

```r
ERROR: 
compilation failed- error
LaTeX Error: Environment abstract undefined.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.328 \begin{abstract}
                       

ERROR: Error
    at Object.onComplete (file:///opt/quarto/bin/quarto.js:92847:91)
    at eventLoopTick (ext:core/01_core.js:153:7)
    at async renderFiles (file:///opt/quarto/bin/quarto.js:78077:16)
    at async renderProject (file:///opt/quarto/bin/quarto.js:78479:25)
    at async Command.actionHandler (file:///opt/quarto/bin/quarto.js:83077:32)
    at async Command.execute (file:///opt/quarto/bin/quarto.js:8017:13)
    at async Command.parseCommand (file:///opt/quarto/bin/quarto.js:7907:20)
    at async quarto (file:///opt/quarto/bin/quarto.js:118224:9)
    at async file:///opt/quarto/bin/quarto.js:118244:9
    at async mainRunner (file:///opt/quarto/bin/quarto.js:118128:9)
Error: Process completed with exit code 1.
```

The error is because my LaTeX document class (krantz) does not include an abstract environment, as noted in the following forum posts:

- https://forum.posit.co/t/quarto-pdf-book-scrbook-environment-abstract-undefined/154982
- https://forum.posit.co/t/latex-error-environment-abstract-undefined/43086

However, it is "by design" that my document class does not include an abstract environment. I don't want the abstract to appear in the PDF version, if possible, because I want to use the krantz document class. Yet I still want to include the Abstract in the HTML version. And I'd prefer not to have to maintain separate `_quarto.yml` files for HTML vs PDF outputs.

To try to address this, I moved `abstract:` to the `format:` section. This allows the book to be knitted to each format--i.e., it no longer results in an error; however, it results in the abstract being printed on every page/chapter of the HTML, as described in the following discussion: https://github.com/quarto-dev/quarto-cli/discussions/2282.

Any help would be greatly appreciated. Basically, I want to accomplish the following:

- I want the HTML output to include an Abstract section, but I do not want to include it at the top of every chapter
- I do **not** want the PDF output to include an Abstract section
- I do **not** to maintain separate `_quarto.yml` files for the HTML, EPUB, and PDF versions
- I do **not** want the PDF version (or any version) to error out when knitting

Thanks in advance!

**Comment by mcanouil** [comment, marked as answer]
Use profiles, see <https://quarto.org/docs/projects/profiles.html>.

**Comment by isaactpetersen** [comment]
This resolved my problem--thanks! For future readers, here's how I accomplished that:

I created a `_quarto-html.yml` file with the following contents:

```yaml
book:
  abstract: INSERT TEXT HERE
```

I removed that code from my global `_quarto.yml` file. I revised the render section of my GitHub Action workflow file to specify my quarto profile:

```yaml
      - name: Render Quarto Project
        uses: quarto-dev/quarto-actions/render@v2
        env:
          QUARTO_PROFILE: html
        with:
          to: html
```
